tests/test_collections.py

```diff
@@ -0,0 +1,313 @@
+        name = self.strip_prefix(self.destFile, self.imageStagingDir)
+        staged_file = os.path.join(self.stagingDirectory, name)
+        self.assertTrue(os.path.exists(staged_file))
+
+        # this file should now not exist
+        self.assertFalse(os.path.exists(self.destFile))
+
+        # add one more task, whose sole purpose is to interrupt the others by
+        # throwing an exception. This is used to exit all tasks.
+        # task_list.append(asyncio.create_task(self.interrupt_me()))
+
+        return staged_file, butler_tasks
+
+    async def testCollectionsTestCase(self):
+        fits_name = "AT_O_20221122_000951_R00_S00.fits.fz"
+        stage_file, butler_tasks = await self.load(fits_name, "collection_test_1.yaml")
+
+        self.check_exposure_count("2020032700020", "LATISS/runs/quickLook", 1)
+        self.check_exposure_count("2022112200951", "LATISS/raw/all", 1)
+
+        task_list = []
+        for butler_task in butler_tasks:
+            task = asyncio.create_task(butler_task())
+            task_list.append(task)
+        task_list.append(asyncio.create_task(self.interrupt_me()))
+        #
+        # kick off all the tasks, until one (the "interrupt_me" task)
+        # throws an exception
+        try:
+            await asyncio.gather(*task_list)
+        except Exception:
+            for task in task_list:
+                task.cancel()
+
+        await asyncio.create_task(self.check_file(stage_file, wait=30, exists=False))
+        self.check_exposure_count("2020032700020", "LATISS/runs/quickLook", 0)
+        self.check_exposure_count("2022112200951", "LATISS/raw/all", 0)
+
+    async def testDoNoDeleteCollectionsTestCase(self):
+        fits_name = "AT_O_20221122_000951_R00_S00.fits.fz"
+        stage_file, butler_tasks = await self.load(fits_name, "collection_test_2.yaml")
+
+        self.check_exposure_count("2020032700020", "LATISS/runs/quickLook", 1)
+        self.check_exposure_count("2022112200951", "LATISS/raw/all", 1)
+
+        task_list = []
+        for butler_task in butler_tasks:
+            task = asyncio.create_task(butler_task())
+            task_list.append(task)
+        task_list.append(asyncio.create_task(self.interrupt_me()))
+        #
+        # kick off all the tasks, until one (the "interrupt_me" task)
+        # throws an exception
+        try:
+            await asyncio.gather(*task_list)
+        except Exception:
+            for task in task_list:
+                task.cancel()
+
+        await asyncio.create_task(self.check_file(stage_file, wait=30, exists=False))
+        self.check_exposure_count("2020032700020", "LATISS/runs/quickLook", 1)
+        self.check_exposure_count("2022112200951", "LATISS/raw/all", 0)
+
+    async def check_file(self, filename, wait=25, exists=True):
+        """Check that the existance of a file
+
+        Parameters
+        ----------
+        filename : `str`
+            name of the file to check
+        wait : `int`
```

Comment:
Please add `optional` to the type specification here and below (for `exists`).

Reply:
Added